### DISCUSSION TITLE
ch2/ch2-02

### DISCUSSION THREAD

**Discussion by giscus** [gopl-zh/gopl-zh.github.com#137] [General]
# ch2/ch2-02

<The Go Programming Language>中文版

https://gopl-zh.github.io/ch2/ch2-02.html

**Comment by mayuhappy7** [comment]
11

**Comment by Cupid7KiKi** [comment]
在 Go 语言中，一个文件中声明的常量是否可以被项目目录下的其他源文件访问，取决于以下几个因素：
1. 常量的访问范围
在 Go 中，常量的访问范围由其 首字母大小写 决定：
首字母大写：常量是 导出的（exported），可以被其他文件访问。
首字母小写：常量是 未导出的（unexported），只能在定义它的文件中访问。
2. 示例
假设你有一个项目目录，包含两个文件：file1.go 和 file2.go。
文件1：file1.go
go复制
package main

const PublicConstant = 42    // 首字母大写，导出的常量
const privateConstant = 100  // 首字母小写，未导出的常量
文件2：file2.go
go复制
package main

import "fmt"

func main() {
    fmt.Println(PublicConstant)  // 可以访问，因为 PublicConstant 是导出的
    // fmt.Println(privateConstant)  // 编译错误：privateConstant 未导出
}
3. 常量的命名要求
大小写：常量的首字母大小写决定了其是否可以被导出。导出的常量首字母必须大写。
命名规范：虽然 Go 没有强制的命名规范，但通常建议使用清晰、有意义的名称。例如：
MaxSize、Pi、DefaultTimeout 等。
包名：如果常量定义在其他包中，需要通过包名访问。例如：
go复制
import "myproject/mypackage"

func main() {
    fmt.Println(mypackage.PublicConstant)
}
4. 总结
导出的常量（首字母大写）可以被项目目录下所有文件访问，只要这些文件属于同一个包或正确导入了包含常量的包。
未导出的常量（首字母小写）只能在定义它的文件中访问。
命名要求：常量的首字母必须大写才能被导出，建议使用清晰的命名方式以提高代码的可读性。
通过合理使用导出和未导出的常量，可以控制常量的访问范围，从而更好地管理项目的代码结构。
